src/mqtt/client.rs

```diff
@@ -122,12 +123,17 @@ impl<'a> Default for MqttClientConfiguration<'a> {
             client_certificate: None,
             private_key: None,
             private_key_password: None,
+
+            #[cfg(all(esp_idf_esp_tls_psk_verification, feature = "alloc"))]
+            psk: None,
         }
     }
 }
 
 #[cfg(esp_idf_version_major = "4")]
-impl<'a> From<&'a MqttClientConfiguration<'a>> for (esp_mqtt_client_config_t, RawCstrs) {
+impl<'a> From<&'a MqttClientConfiguration<'a>>
```

Comment:
This is still NOK, sorry! You are missing the "alloc" protection in the `cfg` block. If `alloc` is not enabled, the `TlsPsk` struct won't be even defined. Please run a build with:
* Both std and alloc disabled
* Alloc enabled only
* Std enabled(which implies alloc)!